CONTRIBUTING.md

```diff
@@ -12,9 +12,85 @@ Make sure the tests pass:
+
+**NOTE:** This endpoint is just an example and doesn't describe the existing `hosts` resource.
+
+Start by adding a new file `hosts.py` to the `datadog/api` folder for the new endpoint. Use the following simple class structure:
+
+```
+from datadog.api.resources import (
+    GetableAPIResource,
+    DeletableAPIResource
+)
+
+class Hosts(GetableAPIResource, DeletableAPIResource):
+    """
+    A wrapper around Hosts HTTP API.
+    """
+    _resource_name = 'hosts'
+```
+
+Each class has the above simple structure, most importantly the following two pieces:
+
+* A `_resource_name` - Indicates the uri of the api.
+* A set of classes to inherit from. This is where the get/post/put/delete request code is defined for you. Available options are:
+  * CreateableAPIResource - Wrapper class for providing a `POST` request for your class, implementing a `create` method.
+  * SendableAPIResource - Fork of CreateableAPIResource class with a `send` method.
+  * UpdatableAPIResource - Wrapper class for providing a `PUT` request for your class, implementing an `update` method.
+  * DeletableAPIResource - Wrapper class for providing a `DELETE` request for your class, implementing a `delete` method.
+  * GetableAPIResource - Wrapper class for providing a `GET` request for your class, implementing a `get` method.
+  * ListableAPIResource - Wrapper class for providing a `GET` request for your class, implementing a `get_all` method.
+  * SearchableAPIResource - Fork of ListableAPIResource class with a `_search` method.
+  * ActionAPIResource - Generic wrapper to trigger any type of HTTP request.
+
+More information about the available classes to inherit from can be found in the [`resources.py`](https://github.com/DataDog/datadogpy/blob/master/datadog/api/resources.py) file.
+
+Looking back at our simple class, we defined the URI this class accesses, `hosts`, and specified that we can call the `delete` and `get` functions by inheriting `GetableAPIResource` and `DeletableAPIResource`.
+
+The remaining piece is to add support for the `GET` request to the `hosts/totals` URI. To do this, update your code to include:
+
+```
+from datadog.api.resources import (
+    GetableAPIResource,
+    DeletableAPIResource,
+    ActionAPIResource
+)
+
+class Hosts(GetableAPIResource, DeletableAPIResource, ActionAPIResource):
+    """
+    A wrapper around Hosts HTTP API.
+    """
+    _resource_name = 'hosts'
+    @classmethod
+    def totals(cls):
+        """
+        Get total number of hosts active and up.
+
+        :returns: Dictionary representing the API's JSON response
+        """
+        return super(Hosts, cls)._trigger_class_action('GET', 'totals')
+```
+
+Notice the addition of the new inherited class `ActionAPIResource`, and the new function `totals`. This new `totals` function calls the `_trigger_class_action` method from that class and appends `totals` to our URI.
```

Comment:
what does the `our URI` refers to? the `hosts`  URI ?

Reply:
Yeah, I'll make that clearer.